### PR TITLE
feat(jwt): allow property to configure JWT Signature Algorithm

### LINKED ISSUE #1744: Add configuration of signature algorithm for JWS
**Is your feature request related to a problem? Please describe.**
There are many options for signature algorithms.

**Describe the solution you'd like**
Add a configuration option to specify the signature algorithm used to create JWS. This will include adding default configuration values, map from string to enum algorithm value, handling misconfiguration, and documentation. Additionally, change default key to a longer value.

### PR DESCRIPTION
**Pull request by bjagg**
Resolves #1744

##### Checklist

-   [x] the [individual contributor license agreement][] is signed
-   [x] commit message follows [commit guidelines][]
-   [x] tests are included
-   [x] documentation is changed or added

##### Description of change

Extract a factory class to supply the configured JWT Signature Algorithm or a default if unable to match/parse property.

[individual contributor license agreement]: https://github.com/Jasig/uPortal/blob/master/CONTRIBUTING.md#individual-contributor-license-agreement
[commit guidelines]: https://github.com/Jasig/uPortal/blob/master/CONTRIBUTING.md#commit
[message properties]: https://github.com/Jasig/uPortal/tree/master/uportal-war/src/main/resources/properties/i18n
[WCAG 2.0 AA]: https://www.w3.org/WAI/WCAG20/quickref/?levels=aaa&technologies=smil%2Cpdf%2Cflash%2Csl

### CODE REVIEW

**Comment by bjagg** [comment]
Wanted to use Lombok for logging. Found that Error Prone needed to be upgraded. Also found that Hibernate metamodel generator was not configured right in a few packages due to my changes.

Still need to test and add a documented entry in portal.properties.

File: uPortal-i18n/build.gradle [hunk 1]
```diff
@@ -23,10 +23,13 @@ sourceSets {
...
             srcDir 'src/main/java'
-            srcDir generatedSourcesDir
         }
     }
-}
+    generated {
+        java {
+            srcDir generatedSourcesDir
+        }
+    }}
```
Comment (bjagg) [inline uPortal-i18n/build.gradle]: This needs to be fixed

File: uPortal-layout/uPortal-layout-core/build.gradle [hunk 2]
```diff
@@ -32,10 +32,13 @@ sourceSets {
...
             srcDir 'src/main/java'
-            srcDir generatedSourcesDir
         }
     }
-}
+    generated {
+        java {
+            srcDir generatedSourcesDir
+        }
+    }}
```
Comment (bjagg) [inline uPortal-layout/uPortal-layout-core/build.gradle]: Yep, anther cut-and-paste error

**Comment by ChristianMurphy** [comment]
Would https://projectlombok.org/features/EqualsAndHashCode or https://projectlombok.org/features/Data be a good fit for the modified DAOs?

**Comment by bjagg** [comment]
@ChristianMurphy Yes! I would love to spend some cycles leveraging more of Lombok. Might replace some of my edits with those as I continue on this PR -- thanks for highlighting those two features. :)

File: uPortal-soffit/uPortal-soffit-core/src/main/java/org/apereo/portal/soffit/service/JwtSignatureAlgorithmFactory.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+ * @since 5.6.1
+ */
+@Component
+@Slf4j
+public class JwtSignatureAlgorithmFactory {
+
+    public static final String SIGNATURE_ALGORITHM_PROPERTY =
+            "org.apereo.portal.soffit.jwt.signatureAlgorithm";
+
+    @Value("${" + SIGNATURE_ALGORITHM_PROPERTY + "}")
```
Comment (jonathanmtran) [inline uPortal-soffit/uPortal-soffit-core/src/main/java/org/apereo/portal/soffit/service/JwtSignatureAlgorithmFactory.java]: Should a default be specified? or will there be an update to the properties file to set a default?

Comment (bjagg) [reply uPortal-soffit/uPortal-soffit-core/src/main/java/org/apereo/portal/soffit/service/JwtSignatureAlgorithmFactory.java]: Yep! Will add a default. It does do a fallback in code, but it should be here as well for consistency

**Comment by bjagg** [comment]
Still need to add a test class around PersonAttributeGroupDefinition recursion of members.

**Comment by asgrant** [comment]
Awesome thanks @bjagg 👍

**Pull request merged**